R/churn.R

```diff
@@ -27,7 +28,8 @@ Churn <- function(data, volume = FALSE, by = "quarter", error.if.no.data = FALSE
     out <-  prop.table(counts, 2)[2, ]
     if (length(out) == 1) # Dealing with scalars losing names
         names(out) <- colnames(counts)
-    dat <- data[data$churn,, drop = FALSE]
+    to <- as.Date(data$to)
+    dat <- data[data$churn & to >= attr(data, "start") & to <= attr(data, "end"),, drop = FALSE]
```

Comment:
There is a strong assumption here that none of the conditions here have missing values. Its probably worth noting that the `data` param given by `RevenueData` is expected to not have any missing values. Or you could use `which(data$churn & to >= attr(data, "start") & to <= attr(data, "end"))` to make the index slightly more robust.